misc/gcd.cpp

```diff
@@ -0,0 +1,44 @@
+/**
+ * @file gcd.cpp
+ * 
+ * @brief 
+ * The idea is, GCD of two numbers doesn’t change if smaller number is subtracted from a bigger number. 
+ * More about  Euclidean algorithm-> https://www.geeksforgeeks.org/euclidean-algorithms-basic-and-extended/
+ * @details 

```

Comment:
Please add more details about the algorithm itself, how it works, etc.. You can take a bit of information from other sources as long as it's not plagiarism.